### PR TITLE
major: Configured app

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
This PR makes the following changes:
- Configured Firebase for the app
- Configured redux toolkit
- Configured routes using React router
- Installed useful packages
- Deployed app to Vercel

Preview link for the changes: https://the-social-network-git-config-app-devansu-yadav.vercel.app/

### CODE REVIEW

File: src/components/Router/PrivateRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useSelector } from "react-redux";
+import { Navigate, useLocation, Outlet } from "react-router-dom";
+
+const PrivateRoute = () => {
+    const location = useLocation();
+    const { authToken } = useSelector((store) => store.auth);
+
+    return authToken ? (<Outlet />): (<Navigate to="/login" state={{ from: location }} replace />);
```
Comment (MarkVed17) [inline src/components/Router/PrivateRoute.jsx:8]: Outlet does wonders! Great use case here.

File: src/redux/slices/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+const authSlice = createSlice({
+    name: "auth",
+    initialState,
+    reducers: {
+
+    }
+});
+
+// export const {} = authSlice.actions;
```
Comment (MarkVed17) [inline src/redux/slices/authSlice.js:15]: you may remove this

**Review by MarkVed17 — APPROVED** [review]
Rest all looks good to merge!

File: src/App.js [hunk 3]
```diff
@@ -1,9 +1,22 @@
 import 'styles/App.css';
+import { AppRoutes } from "components";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.min.css";
 
 function App() {
   return (
     <div className="App">
-      
+      <ToastContainer
```
Comment (abhisheksahani-nova) [inline src/App.js:9]: Suggestion - Use code formatter to arrange code better

**Review by abhisheksahani-nova — APPROVED** [review]
Code is clean and readable, good to merge

**Pull request merged**